### ISSUE TITLE
[Bug]: UI gets stuck after one step of browsing

### ISSUE THREAD

**Issue by neubig** [All-Hands-AI/OpenHands#4790]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

After one step of browsing, the UI gets stuck and doesn't update despite the fact that the agent was still working. Here's a prompt that I used:

```
Please look at this pull request and resolve the remaining issues: https://github.com/All-Hands-AI/openhands-resolver/pull/274
```

After I did this, the UI stopped working after the first agent reply, but after I reloaded the web page the remaining steps in the process were displayed.

### OpenHands Installation

app.all-hands.dev

### OpenHands Version

main

### Operating System

None

**Comment by amanape** [comment]
I'm unable to replicate this bug. I was able to browse through our GitHub repo via OpenHands. [Here](https://www.all-hands.dev/share?share_id=585fa311e58df7eb49b62239e8096fa8c43988a21d98a1e92331660e2dbbbf8a) is the trajectory, but it seems to be loading too much. I guess we never fixed that browser trajectory stuff?

Could you confirm the issue still persists? Maybe it was fixed from a recent merge?

**Comment by enyst** [comment]
The website has troubles opening that trajectory link so I can't see it. 😅 The browser trajectory stuff worked [here](https://github.com/All-Hands-AI/OpenHands/pull/3766), I think, maybe it broke?

**Comment by amanape** [comment]
We've made a change to how feedback is sent in #4597. We use to filter the data to remove the large browser data on the frontend, but seem to have missed that in the new change.

**Comment by neubig** [comment]
Hey @amanape , I was able to replicate the issue in the online version this morning, let me try again.

**Comment by neubig** [comment]
Here's an example (on slack because the movie was too big for github): https://openhands-ai.slack.com/archives/C07S376KASK/p1731357841319049

**Comment by enyst** [comment]
This sounds like the same that I saw tonight on the website. For some reason it was hanging with "Starting up" message _after_ it started browsing. After some refreshes it suddenly displayed what it had done. No error, just, I don't know, slow?

**Comment by neubig** [comment]
I guess this is only a problem online, but this problem is pretty annoying…
I’m also encountering it when not just browsing but also when the agent reads a large log file, etc.

**Comment by mamoodi** [comment]
Is this still happening with the changes to browser?

**Comment by amanape** [comment]
Nope, all fixed!
